shopinvader/components/access_info.py

```diff
@@ -0,0 +1,35 @@
+# Copyright 2019 Camptocamp (http://www.camptocamp.com).
+# @author Simone Orsi <simone.orsi@camptocamp.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo.addons.component.core import Component
+
+
+class PartnerAccess(Component):
+    """Define access rules to partner from client side."""
+
+    _name = "shopinvader.partner.access"
+    _inherit = "base.shopinvader.component"
+    _usage = "access.info"
+    _apply_on = "res.partner"
+
+    def profile(self, partner=None):
+        return {
+            "create": False,
+            "read": True,
+            "update": True,
+            "delete": False,
+            # easy check on client side for being able to edit
+            "readonly": False,
+        }
+
+    def address(self, address_id):
+        return {
+            # TODO: create should be a global params
```

Comment:
at the same time `create` here makes little sense because `create address` permission should be global.
I'd say we should have an`access` or `permission` key in the global payload which states what you can do or not.

[This](https://github.com/shopinvader/odoo-shopinvader/blob/12.0/shopinvader/services/customer.py#L24) seems to be the right place (when you get the customer/profile information).
We could have something like:
```py
    def get(self):
        if self.partner:
            customer = self._get_customer_info()
            return {
                "data": customer,
                "store_cache": {
                    "customer": customer,
                    "permissions": self.access_info.permissions(),
                }
            }
        else:
            return {"data": {}}
```
@lmignon @sebastienbeau what to you think?